libpod/container.go

```diff
@@ -407,6 +409,58 @@ func (c *Container) teardownStorage() error {
 	return nil
 }
 
+func (c *Container) addImageData(g *generate.Generator) (err error) {
+	var (
+		args []string
+	)
+	if !c.config.UseImageConfig {
+		return nil
+	}
+	imageData, err := c.runtime.GetImageConfig(c.config.RootfsImageID)
+	if err != nil {
+		return errors.Wrapf(err, "error reading container image data")
+	}
+	if imageData.User != "" {
+		uid, _, err := chrootuser.LookupUserInContainer(c.bundlePath(), imageData.User)
+		if err != nil {
+			return errors.Wrapf(err, "user %q is not defined in %q", imageData.User, c.bundlePath())
+		}
+		g.SetProcessUID(uint32(uid))
+	}
+	for _, e := range imageData.ExposedPorts {
+		//TODO exposed ports are not handled yet
+		logrus.Warn("Exposed port %q is ignored", e)
+	}
+
+	if len(imageData.Cmd) > 0 {
+		args = imageData.Cmd
+	}
+	if len(imageData.Entrypoint) > 0 {
+		args = imageData.Entrypoint
+	}
+	if len(args) > 0 {
+		g.SetProcessArgs(args)
+	}
+	if imageData.StopSignal != "" {
+		g.AddAnnotation("org.opencontainers.image.stopSignal", imageData.StopSignal)
+	}
+	for _, v := range imageData.Volumes {
+		//TODO volumes are not handled yet
+		logrus.Warn("Builtin Volume %q is ignored", v)
+	}
+	if imageData.WorkingDir != "" {
```

Comment:
Shouldn't set this if it's already set in the spec

Reply:
yup